### PR TITLE
Add Code block

### PR DESCRIPTION
**Pull request by macdonst**
## Description

Adds a Code block component with support for:

- copying the code sample
- syntax highlighting
- line numbers
- highlighted lines

## Related Issue

n/a

## Motivation and Context

Makes our code blocks in our docs look so much nicer.

## How Has This Been Tested?

Added a bunch of storybook tests and tried it in our gatsby project.

## Screenshots (if appropriate):

Here's a kitchen sink screenshot:

Screen Shot 2020-09-10 at 3 34 06 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/Code/index.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  lineNumbers = true,
+  metastring = '',
+  ...props
+}) => {
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false)
+  const language = className.replace(/language-/, '')
+
+  const shouldHighlightLine = calculateLinesToHighlight(metastring)
+  const options = parseMetastring(metastring)
+  const isCopyButton = options.copy !== undefined ? options.copy : copyButton
```
Comment (icaraps) [inline src/Code/index.js:92]: i think you can write `options.copy ?? copyButton` and same below

File: src/Code/index.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  const isLineNumbers =
+    options.numberLines !== undefined ? options.numberLines : lineNumbers
+
+  return (
+    <Provider
+      theme={defaultTheme}
+      colorScheme={theme}
+      scale='medium'
+      UNSAFE_style={{
+        borderRadius: 'var(--spectrum-global-dimension-size-50)'
```
Comment (icaraps) [inline src/Code/index.js:102]: the View component supports setting border-radius so you could remove the UNSAFE_style and wrap the whole thing inside a View

File: src/Code/index.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                      position: fixed;
+                      left: -999px;
+                      opacity: 0;
+                    `}
+                    ref={textarea}
+                    value={children}
+                  />
+                  <ActionButton
+                    aria-describedby={tooltipId}
+                    onClick={() => {
```
Comment (icaraps) [inline src/Code/index.js:133]: it should be `onPress` and you can wrap the ActionButton inside a View since it supports setting absolute positioning.

Comment (macdonst) [reply src/Code/index.js:133]: Switched to onPress. For positioning I'm going to leave it for now. I may have to refactor this to allow the code to wrap around the button anyway.

File: src/Code/index.js (hunk 2 above)
Comment (macdonst) [reply src/Code/index.js:102]: If I don't put the border radius on the Provider element then I don't see the rounded corners. If the View wraps the provider or is wrapped by the Provider I see the border radius is set properly on the div it produces but the Provider is still a black rectangle.

File: src/Code/index.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+const RE = /{([\d,-]+)}/
+
+const calculateLinesToHighlight = (meta) => {
+  if (!RE.test(meta)) {
+    return () => false
+  }
+  const lineNumbers = RE.exec(meta)[1]
+    .split(`,`)
+    .map((v) => v.split(`-`).map((x) => parseInt(x, 10)))
+  console.log(lineNumbers)
```
Comment (icaraps) [inline src/Code/index.js:67]: ^

**Comment by icaraps** [comment]
How does the highlighted lines look like ?

File: src/Code/index.js (hunk 4 above)
Comment (macdonst) [reply src/Code/index.js:67]: Add Docs

**Pull request merged**